sdk/python/feast/infra/online_stores/dynamodb.py

```diff
@@ -0,0 +1,186 @@
+                    table = dynamodb.Table(table_name.name)
+
+        for table_name in tables_to_delete:
+            table = dynamodb.Table(table_name.name)
+            table.delete()
+
+    def teardown(
+        self,
+        config: RepoConfig,
+        tables: Sequence[Union[FeatureTable, FeatureView]],
+        entities: Sequence[Entity],
+    ):
+        online_config = config.online_store
+        assert isinstance(online_config, DynamoDbOnlineStoreConfig)
+        dynamodb = self._initialize_dynamodb(online_config)
+
+        for table_name in tables:
+            try:
+                table = dynamodb.Table(table_name)
+                table.delete()
+            except Exception as e:
+                print(str(e))
+
+    def online_write_batch(
+        self,
+        config: RepoConfig,
+        table: Union[FeatureTable, FeatureView],
+        data: List[
+            Tuple[EntityKeyProto, Dict[str, ValueProto], datetime, Optional[datetime]]
+        ],
+        progress: Optional[Callable[[int], Any]],
+    ) -> None:
+        online_config = config.online_store
+        assert isinstance(online_config, DynamoDbOnlineStoreConfig)
+        dynamodb = self._initialize_dynamodb(online_config)
+
+        table_instance = dynamodb.Table(table.name)
+        with table_instance.batch_writer() as batch:
+            for entity_key, features, timestamp, created_ts in data:
+                document_id = compute_datastore_entity_id(entity_key)  # TODO check id
+                # TODO compression encoding
+                batch.put_item(
+                    Item={
+                        "Row": document_id,  # PartitionKey
+                        "Project": config.project,  # SortKey
+                        "event_ts": str(utils.make_tzaware(timestamp)),
+                        "values": {
+                            k: v.SerializeToString()
+                            for k, v in features.items()  # Serialized Features
+                        },
+                    }
+                )
+
+    def online_read(
+        self,
+        config: RepoConfig,
+        table: Union[FeatureTable, FeatureView],
+        entity_keys: List[EntityKeyProto],
+        requested_features: Optional[List[str]] = None,
+    ) -> List[Tuple[Optional[datetime], Optional[Dict[str, ValueProto]]]]:
+        online_config = config.online_store
+        assert isinstance(online_config, DynamoDbOnlineStoreConfig)
+        dynamodb = self._initialize_dynamodb(online_config)
+
+        result: List[Tuple[Optional[datetime], Optional[Dict[str, ValueProto]]]] = []
+        for entity_key in entity_keys:
+            table_instace = dynamodb.Table(table.name)
+            document_id = compute_datastore_entity_id(entity_key)  # TODO check id
+            response = table_instace.get_item(
+                Key={"Row": document_id, "Project": config.project}
+            )
+            value = response["Item"]
```

Comment:
Should this raise an error instead for one entity key or just prune it? Since this method is for a batch of entity_id, what is the desired behavior for handling errors for each entity?

Reply:
If there's no Item field, it just means that the entity with this id was not found in the online store. In other words it was never materialized. This isn't something we should raise an exception on, since it's expected. In feature_store.py this is handled by assigning the appropriate FieldStatus to the response (FieldStatus.NOT_FOUND): https://github.com/feast-dev/feast/blob/master/sdk/python/feast/feature_store.py#L571